src/replay/service.zig

```diff
@@ -28,100 +28,146 @@ const EpochTracker = replay.trackers.EpochTracker;
 const SlotTracker = replay.trackers.SlotTracker;
 const SlotTree = replay.trackers.SlotTree;
 
+const GossipVerifiedVoteHash = sig.consensus.vote_listener.GossipVerifiedVoteHash;
+const ThresholdConfirmedSlot = sig.consensus.vote_listener.ThresholdConfirmedSlot;
+
 const updateSysvarsForNewSlot = replay.update_sysvar.updateSysvarsForNewSlot;
 
 pub const Logger = sig.trace.Logger("replay");
 
-pub const Service = struct {
-    replay: ReplayState,
-    consensus: ?TowerConsensus,
-    num_threads: u32,
+pub const AdvanceReplayParams = struct {
```

Comment:
Personally I think it's clearer to name this Service or ReplayService and have an advance function on it. That makes it clear that it's the high-level runner for all of replay. Now this just looks like some param struct for some arbitrary function. It's not as clear that this is the sole all-encompassing replay struct.